src/main/scala/is/hail/annotations/OrderedRVIterator.scala

```diff
@@ -13,10 +13,10 @@ case class OrderedRVIterator(
 
   def restrictToPKInterval(interval: Interval): Iterator[RegionValue] = {
     val ur = new UnsafeRow(t.rowType)
-    val pk = new KeyedRow(ur, t.kRowFieldIdx)
+    val pk = new KeyedRow(ur, t.pkRowFieldIdx)
     iterator.filter { rv => {
       ur.set(rv)
-      val keep = interval.contains(t.kType.ordering, pk)
+      val keep = interval.contains(t.pkType.ordering, pk)
```

Comment:
This makes me nervous. The previous version (using keys not partition keys) was definitely *correct*, and if it was doing something wrong, the bug is somewhere else. But given that I'll be changing all of this soon, if this is passing tests and fixes Laurent's problem, I guess it's fine.

Reply:
This can't be correct as it is -- the partition range dependencies are constructed using the partition key, and then this is using the full key. One of the two is definitely wrong, because they're not consistent with each other

Reply:
I think the right fix is probably to change the assertion in `OrderedRVD.apply` from `assert(partitionerBc.value.rangeBounds(i).contains(localType.pkType.ordering, pkUR))` to `assert(partitionerBc.value.rangeBounds(i).contains(localType.kType.ordering, kUR))`. That's what I have in my branch. But it's possible that has other bad interactions with other parts of the code still stuck in PK world. Would you mind giving it a try before we go with this?

Reply:
sure

Reply:
> This can't be correct as it is -- the partition range dependencies are constructed using the partition key, and then this is using the full key. One of the two is definitely wrong, because they're not consistent with each other

They're not actually inconsistent, because the partition key is a prefix of the key. `t.kType.ordering` will compare any two rows of any prefix of the key type correctly, so subsumes `t.pkType.ordering`. And the partition ranges are in general allowed to be any prefix of the key, independently of the partition key. What your change does is makes it effectively truncate the partition bounds to the partition key, but not in the correct way (it doesn't know anything about endpoint inclusivity).

Reply:
I made assumptions about the guarantees of the larger system from those assertions -- if the assertions are wrong, that changes things!

Reply:
in particular, that means that in the current master, the 'partition key' is not really a partition key, because `getPartitionPK()` and `getPartitionK` **do not** need to return the same partition

Reply:
I don't care too much if this is getting fixed soon

Reply:
made the change

Reply:
(still works)